src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -446,20 +461,85 @@ protected void scheduleNext(Callable<Void> task, long delayMillis) {
       logger.debug("Scheduling next token refresh in {} milliseconds", delayMillis);
       try {
         future = firebaseApp.schedule(task, delayMillis);
-      } catch (UnsupportedOperationException ignored) {
+      } catch (UnsupportedOperationException e) {
         // Cannot support task scheduling in the current runtime.
+        logger.debug("Failed to schedule token refresh event", e);
+      }
+    }
+
+    /**
+     * Starts the TokenRefresher if not already started. Starts listening to credentials changed
+     * events, and schedules refresh events every time the OAuth2 token changes. If no active
+     * token is present, or if the available token is set to expire soon, this will also schedule
+     * a refresh event to be executed immediately.
+     *
+     * <p>This operation is idempotent. Calling it multiple times, or calling it after the
+     * refresher has been stopped has no effect.
+     */
+    final synchronized void start() {
+      // Allow starting only from the ready state.
+      if (!state.compareAndSet(State.READY, State.STARTED)) {
```

Comment:
Nit: This might not be needed since state is only ever changed under lock.

Reply:
I'm also using `compareAndSet()` to ensure that we cannot start the refresher once it has been stopped.